R/sample.R

```diff
@@ -34,25 +44,23 @@ mcstate_sample <- function(model, sampler, n_steps, initial = NULL) {
     cli::cli_abort("Expected 'sampler' to be an 'mcstate_sampler'",
                    arg = "sampler")
   }
+  if (is.null(runner)) {
+    runner <- mcstate_runner_serial()
+  }
+  if (!inherits(runner, "mcstate_runner")) {
+    cli::cli_abort("Expected 'runner' to be an 'mcstate_runner'",
+                   arg = "runner")
+  }
 
-  ## We might change this later.
-  rng <- mcstate_rng$new()
-  r_rng_state <- get_r_rng_state()
+  rng <- initial_rng(n_chains)
+  pars <- initial_parameters(initial, model, rng, environment())
+  res <- runner$run(pars, model, sampler, n_steps, rng)
+  combine_chains(res)
+}
 
-  if (is.null(initial)) {
-    ## Really this would just be from the prior; we can't directly
-    ## sample from the posterior!
-    pars <- model$direct_sample(rng)
-  } else {
-    pars <- initial
-    if (length(pars) != length(model$parameters)) {
-      cli::cli_abort(
-        paste("Unexpected initial parameter length {length(pars)};",
-              "expected {length(model$parameters)}"),
-        arg = "initial")
-    }
-  }
 
+mcstate_run_chain <- function(pars, model, sampler, n_steps, rng) {
```

Comment:
I think this function should be in the `runner.R` file, nothing in this file references this